bazel/external/http_parser/http_parser.c

```diff
@@ -2544,6 +2544,70 @@ http_parser_parse_url(const char *buf, size_t buflen, int is_connect,
   return 0;
 }
 
+/* Similar to http_parser_parse_url, but parses the string as
+ * path[?query][#fragment]
+ * */
+int
+http_parser_parse_relative_url(const char *buf, size_t buflen,
```

Comment:
I think this file comes from nodejs/http-parser. Should we instead send a patch upstream instead of carrying a patch here?

Reply:
@bencebeky Is there a better parser in BALSA maybe?

Reply:
nodejs/http-parser is abandoned.  I believe Envoy's copy is identical to the last commit to the external repository.

There is ongoing effort to replace http-parser with BalsaParser, see https://sourcegraph.com/github.com/envoyproxy/envoy@1e9283281f467771b1c6a763224ff81f1ab5d2cb/-/blob/source/common/http/http1/codec_impl.cc?L513-517.  I don't recommend any changes be made to the local copy of http-parser as is will eventually be removed.

If this issue is not specific to HTTP/1, then UHV is probably a better place for tightening up validation logic, because that guarantees uniform behavior across HTTP/1, HTTP/2, and HTTP/3.  See #10646 for more details.

@yanavlasov Can you please advise on whether this logic is better suited to be added to UHV?

Reply:
I agree with @bencebeky that UHV might be a better choice so I'm curious what @yanavlasov thinks.

Reply:
Yes, this should definitely go into UHV. However UHV will be available only in a month or so and it will require use of new codecs. 

If you need this check right now, I suggest implementing it in the `Utility::isValidRefererValue`. You can check the first character, and if it is '/' just check whether there is '#' and additionally whether the character set is valid for URI path.

If it does not start with `/` when use the Http::Utility::URL to parse it as URL and then check for the fragment.

In fact this code will not be used when new codecs are enabled, so I strongly advise against putting it here.